### ISSUE TITLE
episodename shortened to 30 characters

### ISSUE THREAD

**Issue by bartdedecker** [spaam/svtplay-dl#1166]
### svtplay-dl version:

2.4-25-g806fc1d

### Operating system and Python version:

Windows 10
python 3.6.0

### What is the issue:

svtplay-dl shortens the episodename to 30 characters, which sometimes causes duplicate names. If this is meant to be a feature, could you have a config variable (max_filename_len: 30) or an option (--max_filename_len 30)? 

It also misses the fact that the program is one episode of a series and cannot detect season and episode number. The files are placed in the \movies folder instead of a separate folder.

svtplay-dl.yaml:
-----------------
default:
   convert_subtitle_colors: true
   exclude: teckentolkat,teckenspråk,syntolkat
   filename: "{title} - s{season}e{episode} - {episodename}.{ext}"
   get_all_subtitles: true
   path: H:/
   subfolder: true
   subtitle: true
   verbose: true

% svtplay-dl https://www.svtplay.se/video/23859763/fraga-doktorn/fraga-doktorn-sasong-24-25-ar-av-sorg-hon-forlorade-sin-dotter-pa-estonia

INFO: Outfile: H:\movies\fraga-doktorn  - 25-ar-av-sorg-hon-forlorade-si.srt
INFO: Selected to download hls, bitrate: 3100
INFO: Outfile: H:\movies\fraga-doktorn  - 25-ar-av-sorg-hon-forlorade-si.audio.ts
[040/262][=======================.............]
etc.

**Comment by spaam** [comment]
this is related to #1004  and have some discussion in https://github.com/spaam/svtplay-dl/commit/1721ca10030fa1d70aa807c7a30510e10b534189#comments 
what value should we have? 30? more? less?  i dont know.

**Comment by Sopor** [comment]
It shouldn't be any duplicate because the video id is unique for every episode
`fraga-doktorn.25-ar-av-sorg-hon-forlorade-si-91b7fcb-svtplay`

**Comment by Sopor** [comment]
It is still possible to use `-o` to write your own output
```
INFO: Outfile: fraga-doktorn-sasong-24-25-ar-av-sorg-hon-forlorade-sin-dotter-pa-estonia.audio.ts
[262/262][================================================================================] ETA: 0:00:00
INFO: Outfile: fraga-doktorn-sasong-24-25-ar-av-sorg-hon-forlorade-sin-dotter-pa-estonia.ts
[262/262][================================================================================] ETA: 0:00:00
I

**Comment by bartdedecker** [comment]
This does not work if you use the -A option. Then svtplay-dl will generate the names. Some names are long and if season and episode number are not found, and the episodename is long and ends with -part-1 and -part-2, cutting off after a number of characters creates duplicate names and errors are generated: xxx.srt already exists or cannot create xxx.mp4

**Comment by Sopor** [comment]
It should cut off the video id.

**Comment by bartdedecker** [comment]
> this is related to #1004 and have some discussion in [1721ca1#comments](https://github.com/spaam/svtplay-dl/commit/1721ca10030fa1d70aa807c7a30510e10b534189#comments)
> what value should we have? 30? more? less? i dont know.

Why hardcode it? Make it an option or a config variable.

**Comment by Sopor** [comment]
Yes, i wrote it [here](https://github.com/spaam/svtplay-dl/commit/1721ca10030fa1d70aa807c7a30510e10b534189#commitcomment-35301244) as you suggested.

**Comment by spaam** [comment]
@bartdedecker why hardcode it? i'm more or less the only person who fixing bug sadly. because of that i thought it was a good enough solution fixing the issue with some files getting a long name. so i could fix other issues too.  sure it could be a config value.. but that wont fix the issue what the default value should be.

**Comment by iwconfig** [comment]
Yes all parts of the filename should remain but the 30 char limit is too small as i mentioned before. And since a variable limit and switch (for now) doesn't really bring a stable solution, maybe it is better to revert 1721ca1 and just explain the issue to the ones having trouble with too long filenames until a better solution is available. Is this issue even occurring that often? I mean 255 characters is a lot of characters, and I think anything above that would be the result of a parser bug. E.g. the description is mistaken for the episode name (which seems to be the actual issue in #1004).

So I vote for a revert and working towards solving it from the other end.

**Comment by SYSophie** [comment]
If we look at yt-dl there are a few examples on when they limit the filenames to 80 characters. They also have worked on limits around 200 characters. (longer might be preferable in some cases if the system allows it)

Since people have various systems it might be good to look at what name_max the system has and then depending if the string exceeds the max length for the system it should be shortened down.

Another issue that might occur although I think it's much more rare and probably never has happened for any user of svtplay-dl is that the path_max is exceeded by the filename.

It would be nice if there would be checks for these two maxes and that the filename length would be adapted accordingly so that it meets these two requirements.
